src/realm/obj.cpp

```diff
@@ -63,9 +63,9 @@ std::vector<Helper> init_levels(Obj& obj, ColKey origin_col_key, size_t nesting_
     std::vector<Helper> levels;
     levels.reserve(nesting_levels);
 
-    levels.emplace_back(std::move(obj.get_collection_list(origin_col_key)));
+    levels.emplace_back(obj.get_collection_list(origin_col_key));
     for (size_t i = 1; i < nesting_levels; i++) {
-        levels.emplace_back(std::move(levels.back().list->get_collection_list(0)));
```

Comment:
this generates `warning: moving a temporary object prevents copy elision [-Wpessimizing-move]`